modules/govuk_jenkins/spec/classes/jenkins__config_spec.rb

```diff
@@ -1,39 +1,47 @@
-  let(:default_params) {{
-    :github_client_id               => 'bar',
-    :github_client_secret_encrypted => 'baz',
-  }}
+describe "govuk_jenkins::config", :type => :class do
+  let(:default_params) do
+    { github_client_id: "bar", github_client_secret_encrypted: "baz" }
+  end
 
-  describe 'manage config' do
-    context 'false (default)' do
-      let (:params) { default_params.merge({
-        :manage_config => false,
-      })}
+  describe "manage config" do
+    context "false (default)" do
+      let (:params) { default_params.merge(manage_config: false) }
 
-      it { is_expected.not_to contain_file('/var/lib/jenkins/config.xml') }
+      it { is_expected.not_to contain_file("/var/lib/jenkins/config.xml") }
     end
 
-    context 'true' do
-      let (:params) { default_params.merge({
-        :manage_config => true,
-      })}
+    context "true" do
+      let (:params) { default_params.merge(manage_config: true) }
 
-      it { is_expected.to contain_file('/var/lib/jenkins/config.xml').with_content(/<githubWebUri>wibble/) }
+      it { is_expected.to contain_file("/var/lib/jenkins/config.xml").with_content(/<githubWebUri>wibble/) }
     end
 
-    context 'authentication' do
-      let (:params) { default_params.merge({
-        :manage_permissions_github_teams => true,
-        :user_permissions                => [{
-          'user'        => 'admin_user',
-          'permissions' => ['jenkins.perm.admin', 'jenkins.perm.edit'],
-        }],
-      })}
-
-      it { is_expected.to contain_file('/var/lib/jenkins/config.xml').with_content(/<authorizationStrategy class="hudson.security.GlobalMatrixAuthorizationStrategy">/) }
-      it { is_expected.to contain_file('/var/lib/jenkins/config.xml').with_content(/<permission>jenkins.perm.admin:admin_user<\/permission>\n\s+<permission>jenkins.perm.edit:admin_user<\/permission>/) }
+    context "authentication" do
+      let (:params) do
+        default_params.merge(
+          manage_permissions_github_teams: true,
+          user_permissions: [
+            {
+              "user" => "admin_user",
+              "permissions" => ["jenkins.perm.admin", "jenkins.perm.edit"],
+            },
+          ],
+        )
+      end
+
+      it { is_expected.to contain_file("/var/lib/jenkins/config.xml").with_content(/<authorizationStrategy class="hudson.security.GlobalMatrixAuthorizationStrategy">/) }
+      it { is_expected.to contain_file("/var/lib/jenkins/config.xml").with_content(/<permission>jenkins.perm.admin:admin_user<\/permission>\n\s+<permission>jenkins.perm.edit:admin_user<\/permission>/) }
+    end
+
+    context "no new line" do
+      let (:params) { default_params.merge(manage_config: true) }
+
+      it do
+        expect(subject).to contain_file("/var/lib/jenkins/config.xml").without_content(/\n\Z/),
+          "Ensure modules/govuk_jenkins/templates/config/config.xml.erb has no new line. See https://github.com/alphagov/govuk-puppet/pull/11073 for more details."
```

Comment:
```suggestion
          "Ensure modules/govuk_jenkins/templates/config/config.xml.erb does not end with a new line character. See https://github.com/alphagov/govuk-puppet/pull/11073 for more details."
```

I realised this message didn't make much sense as the file can have plenty of new line characters it's just the last one we care about. It seemed surprisingly hard to phrase what we actually mean though so feel free to improve on my suggestion if you have a better idea.

Reply:
I've amended the commit with an improved message - what do you think?

Reply:
Looks a winner 🥇